File: Add Code Here/C++/tree/matrix multiplication.py
p = int(input("enter the column number for matrix1:"))
q = int(input("enter the column number for matrix2:"))

n = int (input("enter the column number for matrix 1/ row number matrix2:"))

# accepting matrix1 values and displaying matrix 1

print ("enter the elements for matrix 1:")

matrix1 = [[int (input ()) for i in range (n)] for j in range (p)]

print ("matrix 1:")

for i in range (p):

    for j in range (n):

        print(format (matrix1[i][j],"<5"), end="")
    
print ()

# accepting matrix2 values and displaying matrix2

print ("enter the elements for matrix2:")

matrix2 = [[int (input()) for i in range (q)] for j in range (n)]

print ("matrix2:")


for i in range (n):

    for j in range (q):

        print (format (matrix2[i][j],"<5"), end="")

print ()

# final matrix

result = [[0 for i in range(q)] for j in range (p)]

for i in range (p) :

    for j in range (q):

        for k in range (n):

            result[i][j] = result[i][j] + matrix1[i][k] * matrix2[k][j]

# displaying the result

print ("the result of matrix multiplication  is:")

for i in range (p):

    for j in range (q):

        print (format (result [i][j], "<5"), end="")

print()







####time complexity is O(n^3)######

File: Add Code Here/C++/Arrays/Matrix Multiplication.cpp
#include <iostream>  
using namespace std;  
int main()  
{  
int a[10][10],b[10][10],mul[10][10],r,c,i,j,k;    
cout<<"enter the number of row=";    
cin>>r;    
cout<<"enter the number of column=";    
cin>>c;    
cout<<"enter the first matrix element=\n";    
for(i=0;i<r;i++)    
{    
for(j=0;j<c;j++)    
{    
cin>>a[i][j];  
}    
}    
cout<<"enter the second matrix element=\n";    
for(i=0;i<r;i++)    
{    
for(j=0;j<c;j++)    
{    
cin>>b[i][j];    
}    
}    
cout<<"multiply of the matrix=\n";    
for(i=0;i<r;i++)    
{    
for(j=0;j<c;j++)    
{    
mul[i][j]=0;    
for(k=0;k<c;k++)    
{    
mul[i][j]+=a[i][k]*b[k][j];    
}    
}    
}    
//for printing result    
for(i=0;i<r;i++)    
{    
for(j=0;j<c;j++)    
{    
cout<<mul[i][j]<<" ";    
}    
cout<<"\n";    
}    
return 0;  
}   
